.github/workflows/publish.yml

```diff
@@ -0,0 +1,69 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Build macOS and Windows
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ main ]
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+*'
+      - 'v[0-9]+.[0-9]+.[0-9]+*'
+
+  pull_request:
```

Comment:
- not sure if this is needed
- i think this is creating the 'refs/pull' releases